ballerina/tests/test.bal

```diff
@@ -50,6 +50,19 @@ function testCreateBucket() {
     }
 }
 
+@test:Config {
+    dependsOn: [testCreateBucket]
+}
+function testCreateObjectWithMetadata() returns error? {
+    map<string> metadata = {
+        "Description" : "This is a text file",
+        "Language" : "English"
+    };
+    
+    Client amazonS3Client = check new(amazonS3Config);
+    _ = check amazonS3Client->createObject(testBucketName, fileName, content, userMetadataHeaders = metadata);
```

Comment:
Shall we test if the metadata was correctly added by getting the object and checking the metadata?

Reply:
The getObject function does not retrieve user-defined metadata. I only updated the createObject function to add user-defined metadata.

Reply:
Can we do something similar to what we did for the doc-ui GET endpoint? Calling the s3 url with range=0 and check the headers of the response?

```
http:Response|error response = httpClient->get("objectUrl", {"Range": "bytes=0-0"});
```

Reply:
This improvement will be tracked in https://github.com/ballerina-platform/ballerina-library/issues/6989